crates/core/src/execution_context.rs

```diff
@@ -162,6 +162,20 @@ impl From<&ReducerContext> for txdata::Inputs {
     }
 }
 
+/// Represents the type of workload that is being executed.
+///
+/// Used as constructor helper for [ExecutionContext].
+#[derive(Clone)]
+pub enum Workload {
+    /// WARNING: This workload type should only be used in tests.
+    ForTests,
```

Comment:
This variant `ForTests` is only here because the old `ExecutionContext::default()` builds a `Address::Zero`.  I think this should be removed and change to `Internal`, but keep this here in case somebody think is usefull

Reply:
Can you `#[cfg(test)]` it?